### PR TITLE
[11.0][MIG] account_move_base_import

### PR DESCRIPTION
**Pull request by grindtildeath**
Migration to v11.0

Included https://github.com/OCA/account-reconcile/pull/252

### CODE REVIEW

File: account_move_base_import/models/account_move.py [hunk 1]
```diff
@@ -262,9 +261,9 @@ def _get_available_columns(self, move_store):
         """Return writeable by SQL columns"""
         model_cols = self._fields
         avail = [
-            k for k, col in model_cols.iteritems() if not hasattr(col, '_fnct')
+            k for k, col in model_cols.items() if not hasattr(col, '_fnct')
         ]
-        keys = [k for k in move_store[0].keys() if k in avail]
+        keys = [k for k in list(move_store[0].keys()) if k in avail]
```
Comment (sebalix) [inline account_move_base_import/models/account_move.py:267]: I'm not sure to understand why you need to call `list` here?

File: account_move_base_import/parser/file_parser.py [hunk 2]
```diff
@@ -79,7 +78,7 @@ def _validate(self, *args, **kwargs):
         separately (in the field: fieldnames).
         """
         if self.fieldnames is None:
-            parsed_cols = self.result_row_list[0].keys()
+            parsed_cols = list(self.result_row_list[0].keys())
```
Comment (sebalix) [inline account_move_base_import/parser/file_parser.py:88]: Same

**Comment by grindtildeath** [comment]
@sebalix This was done using `2to3 -wnj4 --no-diffs .` as explained in https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-11.0#tasks-to-do-in-the-migration, IDK if we should remove it or not :man_shrugging:  

EDIT: I guess it's safer to leave it as it is:
```
>>> {'a': 1, 'b': 2}.keys()
dict_keys(['a', 'b'])
>>> list({'a': 1, 'b': 2}.keys())
['a', 'b']
```

File: account_move_base_import/__init__.py [hunk 3]
```diff
@@ -0,0 +1,4 @@
+
```
Comment (TDu) [inline account_move_base_import/__init__.py]: empty line

File: account_move_base_import/parser/file_parser.py [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+        """
+            :param char: parse_name: The name of the parser
+            :param char: ftype: extension of the file (could be csv, xls or
+              xlsx)
+            :param dict: extra_fields: extra fields to put into the conversion
+              dict. In the format {fieldname: fieldtype}
+            :param list: header : specify header fields if the csv file has no
+              header
+            """
+        super(FileParser, self).__init__(journal, **kwargs)
```
Comment (TDu) [inline account_move_base_import/parser/file_parser.py]: Can be simplify to super()

File: account_move_base_import/parser/generic_file_parser.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+    def __init__(self, journal, ftype='csv', **kwargs):
+        conversion_dict = {
+            'label': ustr,
+            'date': datetime.datetime,
+            'amount': float_or_zero,
+        }
+        # set self.env for later ORM searches
+        self.env = journal.env
+        super(GenericFileParser, self).__init__(
```
Comment (TDu) [inline account_move_base_import/parser/generic_file_parser.py]: Can be simplify to super()

File: account_move_base_import/tests/test_base_completion.py [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+        r"Acsone (.^$*+?()[{\| -]\) SA", r"Line for Acsone (.^$*+?()[{\| -]\) "
+                                         r"SA test", True),
+    name_completion_case("Acšone SA", "Line for Acšone SA test", True),
+]
+
+
+class BaseCompletion(common.TransactionCase):
+
+    def setUp(self):
+        super(BaseCompletion, self).setUp()
```
Comment (TDu) [inline account_move_base_import/tests/test_base_completion.py]: Can be simplify to super()

File: account_move_base_import/tests/test_base_import.py [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+from operator import attrgetter
+from odoo.tests import common
+from odoo import tools
+from odoo.modules import get_module_resource
+
+
+class TestCodaImport(common.TransactionCase):
+
+    def setUp(self):
+        super(TestCodaImport, self).setUp()
```
Comment (TDu) [inline account_move_base_import/tests/test_base_import.py]: Can be simplify to super()

File: account_move_base_import/wizard/__init__.py [hunk 8]
```diff
@@ -0,0 +1,2 @@
+
```
Comment (TDu) [inline account_move_base_import/wizard/__init__.py]: Empty line

File: account_move_base_import/tests/test_invoice.py [hunk 9]
```diff
@@ -0,0 +1,256 @@
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+import datetime as dt
+
+from odoo.modules import get_module_resource
```
Comment (TDu) [inline account_move_base_import/tests/test_invoice.py]: Still works but seems to be deprecated in favor of get_resource_path : https://github.com/odoo/odoo/blob/04d9715766adab97dd6da66c97e50fd6a30dab31/odoo/modules/module.py#L226

File: account_move_base_import/models/account_move.py [hunk 10]
```diff
@@ -274,7 +275,32 @@ def _prepare_insert(self, move, cols):
         move_copy = move
         for k, col in move_copy.items():
             if k in cols:
-                move_copy[k] = self._fields[k].convert_to_column(col, None)
+                # In v11, the record is needed to call convert_to_column on a
+                # Monetary field because it must find the currency to apply the
+                # proper rounding. So we mimic the call here as best as we can.
```
Comment (sebalix) [inline account_move_base_import/models/account_move.py]: Question: if a record is needed, could we use `new()` to get a fake record to retrieve the monetary value and avoid this workaround?

**Review by sebalix — COMMENTED** [review]
One question.
And about 2to3 yes it is safe to leave it like this, but following the number of iterations it could be removed I think to not transform the generator into a list each time.

EDIT: small lint error:
```
account_move_base_import/models/account_move.py:12:1: F401 'odoo.registry' imported but unused
```

**Comment by grindtildeath** [comment]
@sebalix @TDu comments attended

PR rebased to be ready to merge.

**Review by i-vyshnevska — APPROVED** [review]

**Review by TDu — APPROVED** [review]

**Review by florian-dacosta — APPROVED** [review]

**Pull request merged**